scripts/zones/Abyssea-Altepa/npcs/Cruor_Prospector.lua

```diff
@@ -0,0 +1,251 @@
+        Price = 3000;
+        ItemID = 11415;
+    elseif (option == 983041) then -- Teal Pigaches
+        Price = 3000;
+        ItemID = 14258;
+    elseif (option == 17825793) then -- 1x Forbidden Key
+        Price = 500;
+        ItemID = 2490;
+    elseif (option == 84934657) then -- 5x Forbidden Key
+        Price = 2500;
+        ItemID = 2490;
+        Quantity = 5;
+    elseif (option == 168820737) then -- 10x Forbidden Key
+        Price = 5000;
+        ItemID = 2490;
+        Quantity = 10;
+    elseif (option == 504365057) then -- 30x Forbidden Key
+        Price = 15000;
+        ItemID = 2490;
+        Quantity = 30;
+    elseif (option == 839909377) then -- 50x Forbidden Key
+        Price = 25000;
+        ItemID = 2490;
+        Quantity = 50;
+
+    -- Temp items
+    elseif (option == 65538) then -- Lucid Potion I
+        if (player:addTempItem(5824,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5824);
+            player:delCurrency("cruor", 80);
+        end
+    elseif (option == 131074) then -- Lucid Ether I
+        if (player:addTempItem(5827,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5827);
+            player:delCurrency("cruor", 80);
+        end
+    elseif (option == 196610) then -- Catholicon
+        if (player:addTempItem(4206,1)) then
+            player:messageSpecial(ITEM_OBTAINED,4206);
+            player:delCurrency("cruor", 80);
+        end
+    elseif (option == 262146) then -- Dusty Elixer
+        if (player:addTempItem(5433,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5433);
+            player:delCurrency("cruor", 120);
+        end
+    elseif (option == 327682) then -- Clear Salve I
+        if (player:addTempItem(5837,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5837);
+            player:delCurrency("cruor", 120);
+        end
+    elseif (option == 393218) then -- Stalworts Tonic
+        if (player:addTempItem(5839,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5839);
+            player:delCurrency("cruor", 150);
+        end
+    elseif (option == 458754) then -- Ascetics Tonic
+        if (player:addTempItem(5841,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5841);
+            player:delCurrency("cruor", 150);
+        end
+    elseif (option == 524290) then -- Champion's Tonic
+        if (player:addTempItem(5843,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5843);
+            player:delCurrency("cruor", 150);
+        end
+    elseif (option == 589826) then -- Lucid Potion II
+        if (player:addTempItem(5825,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5825);
+            player:delCurrency("cruor", 200);
+        end
+    elseif (option == 655362) then -- Lucid Potion II
+        if (player:addTempItem(5828,1)) then
+            player:messageSpecial(ITEM_OBTAINED,5828);
+            player:delCurrency("cruor", 200);
+        end
+
```

Comment:
stick it into a table, then have a function to handle shit in a global
```
local option_item_map =
{
    {option, items = {{item, price, qty, temp}}, keyitems = {keyitem, price}}, effects = {effect, power}},  
}
```
and have shit call the function, making sure to add keyitem if option_item_map.keyitems ~= nil

i'll double check this comment later and edit as needed